### ISSUE TITLE
Log files can take up to 100gb of disk space

### ISSUE THREAD

**Issue by CapsAdmin** [Heroic-Games-Launcher/HeroicGamesLauncher#1118]
### Describe the bug

This is caused by some bug I guess, but this issue is more that log files are allowed to grow to insane sizes.

This seems to happen if I try to close the launcher. If I close the launcher it hangs and then kde plasma asks if I want to force close it. I guess that leaves some processes hanging.

### Add logs

(01:00:03) INFO:    [Legendary]:       Location: /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/legendary
(01:00:03) INFO:    [Gog]:             Location: /opt/Heroic/resources/app.asar.unpacked/build/bin/linux/gogdl
(01:00:04) INFO:    [Backend]:         
  Heroic Version: 2.2.6 Oden
  Legendary Version:  0.20.25 Our Benefactors
  OS: neon KERNEL: 5.13.0-30-generic ARCH: x64
  CPU: AMD Ryzen 9 5950X 16-Core Processor @3.4 GOVERNOR: ondemand
  RAM: Total: 67.3 GB Available: 45.4 GB
  GRAPHICS: GPU0: Navi 21 [Radeon RX 6900 XT] VRAM: 1MB DRIVER:  
  PROTOCOL: x11
  
(01:00:04) WARNING: [Backend]:         Protocol already registered.
(01:00:04) INFO:    [Frontend]:        Refreshing Library
(01:00:04) INFO:    [Legendary]:       Refreshing library...
(01:00:04) INFO:    [Legendary]:       Refreshing Epic Games...
(01:00:04) INFO:    [Legendary]:       Refreshing Epic Games...
(01:00:05) INFO:    [Legendary]:       [cli] INFO: Logging in...
(01:00:09) INFO:    [Backend]:         checking for heroic updates
(01:00:10) INFO:    [Legendary]:       [Core] INFO: Logging in...
(01:00:10) INFO:    [Legendary]:       [cli] INFO: Getting game list... (this may take a while)
(01:00:11) INFO:    [Legendary]:       Updating game list
(01:00:11) INFO:    [Legendary]:       Game List Updated
(01:00:12) INFO:    [Legendary]:       Checking for game updates
(01:00:12) INFO:    [Legendary]:       Found 0 game(s) to update
(01:00:12) INFO:    [Gog]:             Found 0 game(s) to update

### Steps to reproduce

Maybe start heroic, try to exit, let it hang until kde asks if you want to force exit.

### Expected behavior

some sort of protection against creating log files that are too big in a short time span

### Screenshots

![image](https://user-images.githubusercontent.com/204157/158084982-5f17219d-93d7-4085-a1ea-be3eaba73306.png)
![image](https://user-images.githubusercontent.com/204157/158085155-96d092e3-a68a-49a7-9698-f2eb3810d973.png)

### System Information

kde neon
Not sure what version (heroic --version does not work) but I installed it today using the ppa

**Comment by flavioislima** [comment]
That's weird but I am not sure we can limit the size of logs tbh.
It's also not pretty easy to reproduce this.

**Comment by Nocccer** [comment]
Your PC needs to kill the process correctly. If you don't kill it manually you can get big logs. Heroic can't do anything to kill itself in such state. Actually i remove the old general logs each month. We could remove them each week?

**Comment by CapsAdmin** [comment]
Assuming the error comes from here https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/blob/6c3c7ce941e182c2cd1ea8581da4c0c4147b7d19/electron/main.ts#L440-L443

I can think of some other things to consider:
* Check if the thing you're about to log is identical to the last line of the log, if it is delete the line and append the message with the suffix [x1], [x2], etc.
* Exit if too many exceptions have been caused during a short time
* Exit if the exception is really bad (like this one)

It also seems like a good idea to show the uncaught exception to the user in case there's some bug in the user interface. It's much better to see an error message pop up after clicking a button rather than the button doing nothing.

**Comment by Nocccer** [comment]
Yes we should show the uncaught Exception in the frontend and maybe close heroic. 
The question is what causes this. You are the first person with this problem. Better to fix the problem, instead avoid alot of log messages.

**Comment by Nocccer** [comment]
@CapsAdmin Can you try this appimage and tell me exactly when the exception is thrown?
https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/suites/7653133395/artifacts/318645267

**Comment by CapsAdmin** [comment]
> @CapsAdmin Can you try this appimage and tell me exactly when the exception is thrown? it should now show a error box in frontend and also will close the app after you pressed ok! https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/suites/7653133395/artifacts/318645267

I'm unable to exit this version in an unusual way. The closest thing is getting a white screen when killing one of the sub processes, but that doesn't spam the logs. I never got the message box either by killing sub processes, but I guess that won't cause an immediate issue.

It's been to long to remember how exactly this happened, but I don't think I downloaded an appimage, I think I got it from some ppa on kde neon. Right now I'm on manjaro with kde on a different system, so I can try again later.

**Comment by Nocccer** [comment]
> > @CapsAdmin Can you try this appimage and tell me exactly when the exception is thrown? it should now show a error box in frontend and also will close the app after you pressed ok! https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/suites/7653133395/artifacts/318645267
> 
> I'm unable to exit this version in an unusual way. The closest thing is getting a white screen when killing one of the sub processes, but that doesn't spam the logs. I never got the message box either by killing sub processes, but I guess that won't cause an immediate issue.
> 
> It's been to long to remember how exactly this happened, but I don't think I downloaded an appimage, I think I got it from some ppa on kde neon. Right now I'm on manjaro with kde on a different system, so I can try again later.

The appimage just contains the code i added on #1662. If the EPIPE comes from the uncaught exception handler, we would see where the error is thrown. Maybe [@R0R](https://github.com/R0X0RE0) can try the appimage. He has the same error. Would be good to get the stacktrace.

**Comment by Nocccer** [comment]
Fix is now in heroic 2.4.0.
I will close this for now, if the error still exist, feel free to reopen.

**Comment by cnt0** [comment]
Hello. I just checked, and my `.config/heroic/logs` folder uses more than 50 Gb. I see a lot of errors in log files like this one:
```
09f4:err:combase:RoGetActivationFactory Failed to find library for L"Windows.System.DispatcherQueue"
09f4:fixme:combase:RoGetActivationFactory (L"Windows.System.DispatcherQueue", {00000035-0000-0000-c000-000000000046}, 000000016FA6DDE0): semi-stub
```

```bash
$ cat legendary-2023-10-09T13_32_01.166Z.log | ag DispatcherQueue | wc -l
27915390
```

I'm using heroic 2.9.2 on archlinux. How I can completely disable these logs? Because otherwise it works good, I have no complaints.

**Comment by arielj** [comment]
seems like some game has too much output

you can disable logs in settings > advanced as a workaround

**Comment by cnt0** [comment]
> seems like some game has too much output

should be Fall Guys, that's my most frequent game. Thanks

**Comment by Isaac-Clarke** [comment]
Hiii!

I love necroposting!

Jokes aside, I am "trying out" Death Stranding 2 to see if my Ryzen 5600x, Rx 6650 xt PC can take it, and Heroic is making a ballooning log file which I believe belongs to DS2 ( believe that it's DS2's because if I start the game it modifies the files inside the folder).
I've barely gone through the first level and it has already reached **over 28gb in size.**
I obviously can't upload the log (I can't even open it without the Kernel telling me it "killed Kwrite to avoid a memory thing"), but **the folder is called _4K5wAW2XWTSFph1Kwzou7f_sideload_.**

### I believe that the "dumb fix" to this is to turn logs off, but on the other hand I have no idea what Heroic's "logs" do...

**Comment by Eurico-M** [comment]
>I am "trying out" Death Stranding 2 to see if my Ryzen 5600x, Rx 6650 xt PC can take it, and Heroic is making a ballooning log file which I believe belongs to DS2

Funnily enough, I also installed Death Stranding 2 via Heroic, and my specs are pretty similar (AMD Ryzen 5 5600X, AMD Radeon RX 6600), on CachyOS, and had a folder in ~/.local/state/Heroic/logs/games/ called r8wT61B6euCuiLMhLpo6HH_sideload. 
Inside there are two files, a launch.log (80GB) and a launch.log.old (150GB).

**Comment by Isaac-Clarke** [comment]
> Funnily

tl;dr I tried the game on Windows 10 too and not only it doesn't stutter, but I also get around 10fps more.

idk what the deal is with the "trial" version on Linux via Heroic, but it's making this PILE of data which caches things continuosly!

I may buy the game when it is on sale at the proce **it deserves**, €40, but I believe this must be an Heroic-only problem!
**In fact, after I turned off Logs, the game just stutters and is stuck at 30 fps!**
